src/object/chestSpawner.c

```diff
@@ -1,31 +1,317 @@
+        obj->z.HALF.HI = 0xfff8;
+        ResolveCollisionLayer(obj);
+        obj = CreateFx(obj, FX_DASH, 0);
+        if (obj != NULL) {
+            SoundReq(SFX_124);
+        }
+    }
+}
+
+void ChestSpawner_Type2Action6(ChestSpawnerEntity* this) {
+    if (--super->actionDelay == 0) {
+        super->actionDelay = 8;
+        if (--super->field_0xf != 0) {
+            sub_080840A8(super->x.HALF.HI, super->y.HALF.HI);
+        } else {
+            super->action = 5;
+        }
+    }
+}
+
+void ChestSpawner_Type2Action5(ChestSpawnerEntity* this) {
+    sub_0800445C(super);
+}
+
+void ChestSpawner_Type2Action7(ChestSpawnerEntity* this) {
+}
+
+void ChestSpawner_Type0Init(ChestSpawnerEntity* this) {
+    super->action++;
+    this->tilePosition = COORD_TO_TILE(super);
+    super->hitbox = (Hitbox*)&gUnk_0811F8A8;
+    if (GetTileTypeByEntity(super) == 0x74) {
+        DeleteThisEntity();
+    }
+    if (CheckFlags(this->unk_86)) {
+        super->action = 3;
+        sub_0807B7D8(0x73, this->tilePosition, super->collisionLayer);
+        if ((super->type & 1) == 0) {
+            DeleteThisEntity();
+        }
+    }
+}
+
+void ChestSpawner_Type0Action1(ChestSpawnerEntity* this) {
+    if (CheckFlags(this->unk_86)) {
+        super->action = 2;
+    }
+}
+
+void ChestSpawner_Type0Action2(ChestSpawnerEntity* this) {
+    sub_0807B7D8(0x73, this->tilePosition, super->collisionLayer);
+    switch (super->type) {
+        case 6:
+        case 7:
+            SoundReq(SFX_215);
+            break;
+        default:
+            SoundReq(SFX_SECRET);
+            CreateDust(super);
+            break;
+    }
+    super->action = 3;
+    this->unk_72 = super->actionDelay * 0x3c;
+    if ((super->type & 1) == 0) {
+        DeleteThisEntity();
+    }
+}
+
+void ChestSpawner_Type0Action3(ChestSpawnerEntity* this) {
+    if ((super->type == 1) || (super->type == 7)) {
+        if (GetTileTypeByEntity(super) == 0x74) {
+            DeleteEntity(super);
+        } else {
+            if (!CheckFlags(this->unk_86)) {
+                if (this->unk_72 != 0) {
+                    this->unk_72--;
+                } else {
+                    super->action = 1;
+                    sub_0807BA8C(this->tilePosition, super->collisionLayer);
+                    CreateDust(super);
+                }
+            }
+        }
+    }
+}
+
+void sub_080842D8(ChestSpawnerEntity* this) {
+    static const u8 gUnk_0811F854[] = {
+        35, 64, 191, 255, 35, 64, 192, 255, 38, 64, 255, 255, 38, 64, 0, 0, 255, 255,
+    };
+    static const u8 gUnk_0811F866[] = {
+        39, 64, 191, 255, 35, 64, 192, 255, 40, 64, 193, 255, 41, 64, 255, 255, 38, 64, 0, 0, 42, 64, 1, 0, 255, 255,
+    };
+    static const u8 gUnk_0811F880[] = {
+        34, 64, 255, 255, 34, 64, 0, 0, 255, 255,
+    };
+    static const u8 gUnk_0811F88A[] = {
+        36, 64, 255, 255, 34, 64, 0, 0, 37, 64, 1, 0, 255, 255,
+    };
+    static const u8* const gUnk_0811F898[] = {
```

Comment:
2D array?

Reply:
There is an array with pointers to all these arrays. Can that be compiled from a 2D array?